buildspec-linux-make-gcc-cov.yml

```diff
@@ -0,0 +1,45 @@
+version: 0.2
+
+env:
+  variables:
+    # CodeBuild console doesn't display color codes correctly
+    TESTPL_COLOR_OUTPUT: 0
+
+phases:
+  install:
+    runtime-versions:
+      java: openjdk8
+    commands:
+      - sed -i 's#/archive.ubuntu.com#/us-east-1.ec2.archive.ubuntu.com#g' /etc/apt/sources.list
+      - apt-get update -y
+      - apt-get install -y flex bison make git libwww-perl patch ccache libc6-dev-i386 jq gdb lcov curl python-pip
+      - pip install codecov
+  build:
+    commands:
+      - echo Build started on `date`
+      - make -C src minisat2-download
+      - make -C jbmc/src setup-submodules
+      - make -C src CXX="ccache g++" -j2 CPROVER_WITH_PROFILING=1
+      - make -C unit CXX="ccache g++" -j2 CPROVER_WITH_PROFILING=1
+      - make -C jbmc/src CXX="ccache g++" -j2 CPROVER_WITH_PROFILING=1
+      - make -C jbmc/unit CXX="ccache g++" -j2 CPROVER_WITH_PROFILING=1
+  post_build:
+    commands:
+      - make -C unit test 
+      - make -C regression test CPROVER_WITH_PROFILING=1
+      - make -C regression/cbmc test-paths-lifo
+      - env PATH=$PATH:`pwd`/src/solvers make -C regression/cbmc test-cprover-smt2
+      - make -C jbmc/unit test
+      - make -C jbmc/regression test
+      - lcov --capture --directory . --output-file ./lcov.info
+      - VCS_PULL_REQUEST=$(echo $CODEBUILD_SOURCE_VERSION | sed 's/pr\///g')
+      - COV_SCRIPT=/root/.cache/codecov.sh
+      - if [ ! -f "$COV_SCRIPT" ]; then curl -s https://codecov.io/bash > "$COV_SCRIPT"; fi
+      - echo "$CODEBUILD_INITIATOR" | grep GitHub && bash "$COV_SCRIPT" -t "$CODECOV_TOKEN" || true      
```

Comment:
I don't fully understand what this line is doing (I'm afraid I don't know what the contents of `$CODEBUILD_INITIATOR` is off the top of my head) - but I'm slightly worried by the `|| true` - which will basically mask any errors that occur here?

Reply:
Well, essentially I blindly used script example here 
https://github.com/codecov/codecov-python/issues/140
According to https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html, `$CODEBUILD_INITIATOR` is "the entity that started the build". In our case, (also from log), I believe this is webhook? 
`[Container] 2019/05/30 09:35:34 Running command echo "$CODEBUILD_INITIATOR" | grep GitHub && bash "$COV_SCRIPT" -t "$CODECOV_TOKEN" || true 
GitHub-Hookshot/0a2cefb `

Do you want me to try removing the `|| true`?

Reply:
I'm honestly not sure on the `|| true` - I guess lets run it as is and see how reliable it is.